### ISSUE TITLE
Elasticsearch-py 6 doesn't work for Elasticsearch6 upgraded from Elasticsearch5, but Elasticsearch-py 5 works.

### ISSUE THREAD

**Issue by jh88** [elastic/elasticsearch-py#796]
My cluster version: 
```
'version': {'number': '6.2.4', 'build_hash': 'ccec39f', 'build_date': '2018-04-12T20:37:28.497551Z', 'build_snapshot': False, 'lucene_version': '7.2.1', 'minimum_wire_compatibility_version': '5.6.0', 'minimum_index_compatibility_version': '5.0.0'}
```

```
from elasticsearch import Elasticsearch
es = Elasticsearch(['https://user:password@my.host:9243'])
```
I got this error:

> ImproperlyConfigured("Root certificates are missing for certificate "
> elasticsearch.exceptions.ImproperlyConfigured: Root certificates are missing for certificate validation. Either pass them in using the ca_certs parameter or install certifi to use it automatically.

However Elasticsearch-py5 works fine.

**Comment by fxdgear** [comment]
Have you installed `certifi`?

**Comment by jh88** [comment]
@fxdgear Yes I did

> certifi                   2018.1.18                py36_0

I only has this error when I try to connect to Elasticsearch 6 upgraded from 5. It works fine with Elasticsearch6 if it's installed directly.

**Comment by fxdgear** [comment]
> I only has this error when I try to connect to Elasticsearch 6 upgraded from 5. It works fine with Elasticsearch6 if it's installed directly.


If I'm understanding what you're trying to say: 
* you have elasticsearch5 installed 
* you did a `pip install elasticsearch` to install the lastest version
* then things broke?

If you don't install `certifi` using SSL will fail, if you don't provide any CA data yourself. 

`certifi` isn't installed by default. It's an optional dependency. 

If your ES cluster is using self-signed certs certifi won't help you because it doesn't have any CA data for self signed certs. 
If your ES cluster is using pubicly create (lets-encrypt or some other "for-pay" signing service) then Certifi will work, but you have to install that dependency yourself. 

i hope this helps
thanks

**Comment by jh88** [comment]
@fxdgear 
This's what I'm trying to say:
1. My elasticsearch6 cluster was upgraded from elasticsearch5
2. My elasticsearch-py6 was installed by `conda install -c conda-forge elasticsearch`, and it works fine for elasticsearch6 clusters which are installed directly, i.e. not upgraded from elasticsearch5. However it didn't work for the cluster upgraded from elasticsearch5.
3. I also have an old elasticsearch-py5 in other virtual environments, and it works fine for the elasticsearch6 cluster upgraded from elasticsearch5

I doubt it's because `certifi`, because if you try to use elasticsearch-py6 to connect an elasticsearch5 cluster, you will get the same error.

**Comment by jh88** [comment]
The problem is fixed after upgrading to elasticsearch-py 6.2
